src/node/audio_buffer_source.rs

```diff
@@ -49,9 +48,55 @@ struct PlaybackInfo {
     k: f32,
 }
 
+// The strategy for loop parameters is as follow: store the given values
+// in the `loop_parameters` thread safe instance which only lives in control
+// thread(s) and send a message to the render thread which stores the raw values.
+// Values between control and render side might be desynchronised for little while
+// but the developer experience will appear more logical, i.e.
+// ```no_run
+// node.set_loop(true);
+// println!("{:?}", node.loop_());
+// > true // is guaranteed
+// ```
+// Note that this seems to be the strategy used by Firefox
+#[derive(Debug)]
+struct LoopParameters {
```

Comment:
How about renaming this to `LoopControl` because it is a stateful, mutable container? The name `LoopParameters` suggests that this is just a simple value object, i.e. a POD.

Reply:
ok done